x/concentrated-liquidity/model/pool_test.go

```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
 	suite.Run(t, new(ConcentratedPoolTestSuite))
 }
 
+// TestGetAddress tests the GetAddress method of pool
+func (s *ConcentratedPoolTestSuite) TestGetAddressAndGetIncentiveAddress() {
+
+	tests := []struct {
+		name          string
+		expectedPanic bool
+	}{
+		{
+			name: "Happy path",
+		},
+		{
+			name:          "Unhappy path: wrong bech32 encoded address",
+			expectedPanic: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a concentrated liquidity pool struct instance
+			address := s.TestAccs[0].String()
+
+			// if the test case is expected to panic, we use wrong bech32 encoded address instead
+			if tc.expectedPanic {
+				address = "osmo15l7yueqf3tx4cvpt6njvj7zxmvuhkwyrr509e9"
+			}
+			mock_pool := model.Pool{
+				Id:      1,
+				Address: address,
+			}
+
+			// Check that the returned address is backward compatible
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				addr := mock_pool.GetAddress()
+				s.Require().Equal(addr, s.TestAccs[0])
+			})
+		})
+	}
+}
+
+// TestGetIncentivesAddress tests the GetIncentivesAddress method of pool
+func (s *ConcentratedPoolTestSuite) TestGetIncentivesAddress() {
+
+	tests := []struct {
+		name          string
+		expectedPanic bool
+	}{
+		{
+			name: "Happy path",
+		},
+		{
+			name:          "Unhappy path: wrong bech32 encoded address",
+			expectedPanic: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a concentrated liquidity pool struct instance
+			address := s.TestAccs[0].String()
+
+			// if the test case is expected to panic, we use wrong bech32 encoded address instead
+			if tc.expectedPanic {
+				address = "osmo15l7yueqf3tx4cvpt6njvj7zxmvuhkwyrr509e9"
+			}
+			mock_pool := model.Pool{
+				Id:      1,
+				Address: address,
+			}
+
+			// Check that the returned address is backward compatible
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				addr := mock_pool.GetAddress()
+				s.Require().Equal(addr, s.TestAccs[0])
+			})
```

Comment:
This isn't getting the incentives address

Reply:
Fixed test to do so in https://github.com/osmosis-labs/osmosis/pull/5266/commits/51fce1f3393cc2e43e9de0e707d86ad5ca561b34